packages/components/codemods/migrateNotificationTypeToVariant/migrateNotificationTypeToVariant.ts

```diff
@@ -0,0 +1,52 @@
+import ts from "typescript"
+import { getPropValueText } from "../utils/getPropValueText"
+
+/** Recurses through AST to find and update any jsx element that matched the tagName */
+export const transformNotificationTypeToVariant =
+  (context: ts.TransformationContext, tagName: string) =>
+  (rootNode: ts.Node): ts.Node => {
+    function visit(node: ts.Node): ts.Node {
+      if (ts.isJsxOpeningElement(node) || ts.isJsxSelfClosingElement(node)) {
+        if (node.tagName.getText() === tagName) {
+          const newAttributes = node.attributes.properties.map(attr => {
+            if (ts.isJsxAttribute(attr) && attr.name.getText() === "type") {
+              const valueName =
+                attr.initializer && getPropValueText(attr.initializer)
+
+              if (valueName) {
+                let variantValue: string = valueName
+
+                // Only change values that are necessary
```

Comment:
Nit: Not necessary to change I suppose given you have tests, but I think it's actually easier to understand and not look like cases are missing if we list all the options in the switch.